src/go/rpk/pkg/cli/registry/schema/schema.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package schema
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/sr"
+)
+
+func NewCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "schema",
+		Args:  cobra.ExactArgs(0),
+		Short: "Manage your schema registry schemas",
+	}
+	cmd.AddCommand(
+		newCheckCompatibilityCommand(fs, p),
+		newCreateCommand(fs, p),
+		newDeleteCommand(fs, p),
+		newGetCommand(fs, p),
+		newListCommand(fs, p),
+	)
+	p.InstallFormatFlag(cmd)
+	return cmd
+}
+
+type subjectSchema struct {
+	Subject string `json:"subject" yaml:"subject"`
+	Version int    `json:"version" yaml:"version"`
+	ID      int    `json:"ID" yaml:"ID"`
+	Type    string `json:"type" yaml:"type"`
+}
+
+func printSubjectSchemaTable(f config.OutFormatter, single bool, ss ...sr.SubjectSchema) error {
+	var (
+		rows   []subjectSchema
+		tw     *out.TabWriter
+		isText = f.IsText()
+	)
+	if isText {
+		tw = out.NewTable("subject", "version", "id", "type")
+		defer tw.Flush()
+	}
+	for _, s := range ss {
+		r := subjectSchema{
+			Subject: s.Subject,
+			Version: s.Version,
+			ID:      s.ID,
+			Type:    s.Type.String(),
+		}
+		if isText {
+			tw.PrintStructFields(r)
```

Comment:
It might be a bit of duplication but can all of the output printing come at the end, and right here we just build rows?

Reply:
This is on purpose:

If is **text** you just want to print the row, if it's json/yaml you'll want to have the full slice and then `f.Format()`

In my first pass, I had a first for loop building the rows, and at the end, another for loop for printing. But this isn't optimal for users that have a lot of schemas since we'll have to iterate through the array twice.

Reply:
The time delay of iterating these rows twice is incomparably small in comparison to fetching the array and printing it, so much so that IMO we should structure the code around readability.